src/core/Global.ttslua

```diff
@@ -825,3 +779,146 @@ function urldecode(str)
+  
+  -- option: Show token arranger
+  elseif id == "showTokenArranger" then
+    -- delete previously pulled out tokens
+    for _, token in ipairs(getObjectsWithTag("to_be_deleted")) do token.destruct() end
+
+    spawnOrRemoveHelper(state, "Token Arranger", {-42.3, 1.4, -46.5})
+
+  -- option: Show clean up helper
+  elseif id == "showCleanUpHelper" then
+    spawnOrRemoveHelper(state, "Clean Up Helper", {-68, 1.6, 35.5})
+  
+  -- option: Show hand helper for each player
+  elseif id == "showHandHelper" then
+    spawnOrRemoveHelper(state, "Hand Helper", {-50.84, 1.6, 7.02}, {0, 270, 0}, "White")
+    spawnOrRemoveHelper(state, "Hand Helper", {-50.90, 1.6, -25.10}, {0, 270, 0}, "Orange")
+    spawnOrRemoveHelper(state, "Hand Helper", {-34.38, 1.6, 22.44}, {0, 000, 0}, "Green")
+    spawnOrRemoveHelper(state, "Hand Helper", {-16.69, 1.6, -22.42}, {0, 180, 0}, "Red")
+
+  -- option: Show chaos bag manager
+  elseif id == "showChaosBagManager" then
+    spawnOrRemoveHelper(state, "Chaos Bag Manager", {-67.8, 1.4, -49.5})
+
+  end
+end
+
+-- handler for spawn / remove functions of helper objects
+---@param state Boolean Contains the state of the option: true = spawn it, false = remove it
+---@param name String Name of the helper object
+---@param position Vector Position of the object (where it will spawn or where it will be removed from)
+---@param rotation Vector Rotation of the object for spawning (default: {0, 270, 0})
+---@param color Color This is only needed for correctly setting the color of the "Hand Helper"
+function spawnOrRemoveHelper(state, name, position, rotation, color)
+  if state then
+    spawnHelperObject(name, position, rotation, color)
+    Player["White"].pingTable(position)
+  else
+    removeHelperObject(name, position)
+  end
+end
+
+-- copies the specified tool (by name) from the barrel
+---@param name String Name of the object that should be copied
+---@param position Position Desired position of the object
+function spawnHelperObject(name, position, rotation, color)
+  if rotation == nil then rotation = {0, 270, 0} end
+
+  local barrel = getObjectFromGUID("aa8b38")
+  local barrelCopy = barrel.clone({position = barrel.getPosition() + Vector(0, 3, 0)})
+
+  for _, obj in ipairs(barrel.getObjects()) do
+    if obj.name == name then
+      barrelCopy.takeObject({
+        index = obj.index,
+        position = position,
+        rotation = rotation,
+        callback_function = function(object)
+          if name == "Hand Helper" then
+            Wait.time(function() object.call("externalColorChange", color) end, 1)
+          end
+        end,
+        smooth = false
+      })
+      barrelCopy.destruct()
+      return
+    end
+  end
+end
+
+-- removes the specified tool (by name) from the provided position
+---@param name String Name of the object that should be removed
+---@param position Position Position of the object
+function removeHelperObject(name, position)
+  local search = Physics.cast({
```

Comment:
You only look for the helper at the original spawn position, so this will fail if the user moves it, right?  That actually seems like a pretty common occurrence.

Could fix this by storing the GUID(s) of spawned objects as the value of the table (rather than just a boolean).  nil = off, string/string array = on

Reply:
Yes, that's how I implemented it at the moment.

Storing the GUID would be possible, but only removing from the default position feels a bit safer to me.
Otherwise the custom set position would be gone.

Reply:
If that happens though, then the option panel doesn't do what it says.  If a user disables an option, it would remain in play and the operation not only fails but the presentation is inconsistent.  And if they re-enable it, now they've got multiples :)

I don't think "Turning this on places it at X and then you can move it" is the worst thing.  IMHO the user signal to remove the object is much stronger than repositioning it, and if it is done in error moving it back where they want it is easy for them.

If keeping the position is important, can always store the position before destroying, and spawn the new one there if they re-enable it.  Though with our inability to persist anything across new versions/saves, the object appearing at its default spot is going to be the behavior most of the time anyway, so there's probably a consistency win.

Reply:
Storing the GUID of the spawned object is the best solution imo.
Since that needs a change to the save function, I'd like to mvoe that into the next part.